src/source/raster_dem_tile_source.js

```diff
@@ -0,0 +1,128 @@
+// @flow
+
+const ajax = require('../util/ajax');
+const util = require('../util/util');
+const Evented = require('../util/evented');
+const normalizeURL = require('../util/mapbox').normalizeTileURL;
+const browser = require('../util/browser');
+const TileCoord = require('./tile_coord');
+const RasterTileSource = require('./raster_tile_source');
+const {deserialize} = require('../util/web_worker_transfer');
+
+import type {Source} from './source';
+import type Dispatcher from '../util/dispatcher';
+import type Tile from './tile';
+import type {Callback} from '../types/callback';
+
+
+class RasterDEMTileSource extends RasterTileSource implements Source {
+    _options: RasterSourceSpecification;
+
+    constructor(id: string, options: RasterSourceSpecification, dispatcher: Dispatcher, eventedParent: Evented) {
+        super(id, options, dispatcher, eventedParent);
+        this.type = 'raster-dem';
+        this.maxzoom = 15;
```

Comment:
Why is the maxzoom hard coded?

Reply:
terrain-rgb tiles stop at z14, but I changed it to 22 for consistency with other source types.